### DISCUSSION TITLE
Can EdgeUpdate be enabled through the Edge process to install WebView2

### DISCUSSION THREAD

**Discussion by sln162** [MicrosoftEdge/WebView2Feedback#4031] [Q&A]
It's great to see the announcement that WebView2 has started to be pushed to all Win10 devices, but there are still some circumstances that limit its installation.
According to my current test, WebView2 is installed on the Win10 machine through Edge's task plan/service (the actual test should be task plan), but the installed antivirus software (such as 360) will prompt when installing Edge Disable all task scheduling and services, even if they are installed first, they will be disabled by default when the antivirus software's startup items are optimized.

After I tested and disabled, WebView2 could not be installed automatically. Edge Update is installed along with Edge and generally exists.
Can the Edge Update process be invoked when the user browses the web with Edge to implement the silent installation of WebView2 in the background.

The following picture shows that it is recommended to disable these by default, but if WebView2 has not been installed or is uninstalled, it will not be installed automatically,back to the title, Can EdgeUpdate be enabled through the Edge process to install WebView2?

![image](https://user-images.githubusercontent.com/11196986/226151057-014ee0b3-ce2a-4506-8462-fd294e67eed3.png)
![image](https://user-images.githubusercontent.com/11196986/226151065-e8a6115f-2516-4e31-b472-9d8a38d5843d.png)

**Comment by novac42** [comment]
Thanks for reaching out. I've assigned this to a dev that can help follow up on this.

**Comment by LiangTheDev** [comment]
When Edge Update tasks are disabled, all products managed by it would not get updated, including Edge browser. So, it is currently not possible to let the running Edge browser to trigger installation of WebView2 Runtime. If the user uses Edge browser, it is not a good idea to disable Edge Update as that would stop Edge browser from getting security updates and make it unsafe.

**Comment by sln162** [comment]
>  If the user uses Edge browser, it is not a good idea to disable Edge Update as that would stop Edge browser from getting security updates and make it unsafe.

As ordinary users, they don't care about these tasks. They rely on antivirus software and habitually click "one click optimization", while antivirus software prohibits these tasks by default.

Even if the update task is disabled, if the user manually clicks "About Microsoft Edge", the Edge should also be able to automatically update when entering this page. Can WebView2 be installed at this time (on machines lacking WebView2)? If not, can you convey this suggestion to the Edge installation team? Thank you.

**Comment by LiangTheDev** [comment]
We have discussed this with installer team. Yes, "About Microsoft Edge" can trigger updates of the Edge browser, that is just for the Edge browser. The best way is to talk to 360. Those scheduled update tasks are not running at startup and should not impact startup perf and disabling it will impact security a lot.

**Comment by sln162** [comment]
[360 forum](https://bbs.360.cn/forum.php?mod=viewthread&tid=16080127)
Some people have reported this problem, 360 official said it will not deal with it
